### ISSUE TITLE
Add option to see both per window and global count at the same time

### ISSUE THREAD

**Issue by yphastos** [lllusion3418/webext_tab_number#1]
It would be useful to have an option to see both per window and global count at the same time.
I think both counters could fit perfectly in the icon, by making the text smaller, and in two rows, e.g.

> 
> 14/
> 23

So we can get the full picture at a glance.

**Comment by lllusion3418** [comment]
This is what it could look like (64px):
![Screenshot_20190519_202423](https://user-images.githubusercontent.com/4203927/57986366-44c3fa00-7a74-11e9-856b-7521d5cd3d32.png)

A few questions:
* do we want to support different colors for the different rows?
* what about the length restriction for the badge? it's only 4 characters
* do we want to support reordering?
* maybe add additional categories (e.g. active tabs)

Edit:
* what about the case when there's only one window?

**Comment by lllusion3418** [comment]
Right now, I've implemented it like this:
> do we want to support different colors for the different rows?

no

> what about the length restriction for the badge? it's only 4 characters

don't allow "both" in badge mode

> do we want to support reordering?

no

> maybe add additional categories (e.g. active tabs)

no

> what about the case when there's only one window?

only show single number

**Comment by yphastos** [comment]
All of those options seem perfect to me. 

At first I also thought of including a window counter, but it would be confusing and it is not very useful anyway.

**Comment by lllusion3418** [comment]
you can test it here: https://github.com/lllusion3418/webext_tab_number/releases/tag/v0.9-alpha2

it also includes another change which I just hadn't published yet: a font selection gui (sort of)

**Comment by yphastos** [comment]
Thanks! It works, but both numbers appear too close to each other (cramped), and since the text is small, is a bit hard to read. It should have a bit of space (padding?) between the two rows. A couple pixels would do i think.

Edit, heres a screenshot:

![image](https://user-images.githubusercontent.com/13773388/58040013-f78a6b80-7af9-11e9-936e-ccfb0969591f.png)


Changing the icon size helps a bit. Seemingly improves anti-aliasing and appears more readable, but still they are too close.

I also tried the margin option at 0% to make text bigger, but it makes no difference. It still has a margin for the whole icon. I guess that is a restriction of the icons themselves in firefox, not the addon.

Changing the font may help (the new GUI works fine btw), making thinner numbers, but since the vertical space doesn't change, they still look cramped.

**Comment by lllusion3418** [comment]
The way it's implemented right now, "margin" is the fraction of the whole height that is (supposed to be) kept empty at the top, the bottom, and between the two rows.

E.g if the icon dimensions are 64px and margin is set to 5%, there's supposed to be 3.2px emptiness, 27.2px text, 3.2px emptiness, 27.2px text, and then 3.2px emptiness again

The problem is that it's not possible to tell the HTML canvas API to make a digit exactly `X` px high - you can only tell it a certain font size (in px, em, pt, or whatever). But how exactly that relates to the height of the digits/numbers we're interested in is entirely up to the font you're using. (at least as far as I can tell)

The same goes for the baseline (that is the bottom of the text). It could be directly underneath of our number, but it could also be anywhere else. (regardless of whether you're using "alphabetic", "ideographic", or "bottom")

This the reason why [this code](https://github.com/lllusion3418/webext_tab_number/blob/b2063c0ff06c953d5b9cf20fb003e2090321571c/background.js#L362-L412) is there.

Edit: At least that's how I remember it - it's been almost 2 years since I last touched that code.

**Comment by yphastos** [comment]
I see. I thought margin was only 'around' the text (applied to both rows as if they were one entity), and not between, that's why I tried setting it to 0%, attempting to maximize overall size of both rows.

According to what you said, and messing around a bit with the options, I found that the best results are with an icon size of 32, monospace font, and a margin of 7.5%, which gives this:

![image](https://user-images.githubusercontent.com/13773388/58055499-88c20800-7b23-11e9-9e34-f2ca04e94e8e.png)

It looks good enough for me! Thank you.
Let me know if I can help any further.
